File: test-suite/src/test/java/io/micronaut/docs/metrics/configurer/SimpleMeterRegistryConfigurerTest.java
package io.micronaut.docs.metrics.configurer;

import io.micrometer.core.instrument.MeterRegistry;
import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertInstanceOf;

@Property(name = "spec.name", value = "SimpleMeterRegistryConfigurerTest")
@MicronautTest
class SimpleMeterRegistryConfigurerTest {

    @Inject
    MeterRegistry meterRegistry;

    @Test
    void theConfigurerIsAppliedToTheSimpleMeterRegistry() {
        // the SimpleMeterRegistry is the default registry of the composite when no other registry is configured
        var composite = assertInstanceOf(CompositeMeterRegistry.class, meterRegistry);
        var simpleMeterRegistry = composite.getRegistries().stream()
                .filter(SimpleMeterRegistry.class::isInstance)
                .findFirst()
                .orElseThrow();

        meterRegistry.counter("configurer.test").increment();

        var counter = simpleMeterRegistry.get("configurer.test").tags("key", "value").counter();
        assertEquals(1.0, counter.count());
    }
}

File: test-suite-python/src/test/python/micronaut/docs/metrics/configurer/SimpleMeterRegistryConfigurerTest.py
from typing import Annotated

from io.micrometer.core.instrument import MeterRegistry
from io.micrometer.core.instrument.simple import SimpleMeterRegistry
from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@Property(name="spec.name", value="SimpleMeterRegistryConfigurerTest")
@MicronautTest
class SimpleMeterRegistryConfigurerTest:
    meterRegistry: Annotated[MeterRegistry, Inject]

    @Test
    def test_the_configurer_is_applied_to_the_simple_meter_registry(self):
        # the SimpleMeterRegistry is the default registry of the composite when no other registry is configured
        simpleMeterRegistry = None
        for registry in self.meterRegistry.getRegistries():
            if isinstance(registry, SimpleMeterRegistry):
                simpleMeterRegistry = registry
        assert simpleMeterRegistry is not None

        self.meterRegistry.counter("configurer.test").increment()

        counter = simpleMeterRegistry.get("configurer.test").tags("key", "value").counter()
        assert counter.count() == 1.0
